eng/build.sh

```diff
@@ -165,6 +165,9 @@ while [[ $# > 0 ]]; do
     --sourcebuild)
       source_build=true
       ;;
+    /p:ArcadeBuildFromSource=true)
```

Comment:
Do we need to pass this property along to the later msbuild calls? Looks like it get swallowed here.

Reply:
It sets the `source_build` property, which ends up getting translated back into this same command.

Reply:
At least it was _supposed_ to set the source_build property, but from the build output it appears to do nothing at all.

Reply:
```suggestion
    /p:arcadebuildfromfource=true)
```

The input is lowercased at the start of the switch.

Reply:
I would also consolidate this with the previous pattern as well, with a comment:

```bash
--sourcebuild|/p:arcadebuildfromsource=true)
  # Arcade specified ArcadeBuildFromSource, not --sourcebuild, but that's not developer friendly so we have an alias.
```

Reply:
Thanks @333fred these are great improvements 👍

Reply:
Actually, @MichaelSimmons does this flag need to be passed all the way through our build script? The pipeline is passing, but I see a number of places in arcade that reference this property.

Reply:
The source-build pipeline is passing in /p:ArcadeBuildFromSource=true to the build script.  The `ArcadeBuildFromSource` flag must be passed through the build script to msbuild.

Reply:
Alright. @sharwell, we'll need to update the build script to set that property.

Reply:
We do that right?
https://github.com/dotnet/roslyn/blob/2e0058d14cf9cb4f0140255314b1822ce7c21b63/eng/build.sh#L309

Reply:
Ah, so we do. Thanks Sam.